// @strapi/design-system/dist/CarouselInput/CarouselInput.js
import r from "react";
import e from "prop-types";
import { Carousel as R } from "./Carousel.js";
import { Field as E } from "../Field/Field.js";
import { FieldLabel as y } from "../Field/FieldLabel.js";
import "../Field/FieldInput.js";
import { FieldHint as L } from "../Field/FieldHint.js";
import { FieldError as F } from "../Field/FieldError.js";
import "../Field/FieldContext.js";
import "../Field/FieldAction.js";
import { Stack as h } from "../Stack/Stack.js";
import { useId as x } from "../helpers/useId.js";
const t = ({
  actions: n,
  children: d,
  error: l,
  hint: s,
  label: i,
  labelAction: a,
  nextLabel: m,
  onNext: u,
  onPrevious: p,
  previousLabel: c,
  required: f,
  secondaryLabel: g,
  selectedSlide: q,
  id: b,
  ...v
}) => {
  const o = x("carouselinput", b);
  return /* @__PURE__ */ r.createElement(E, {
    hint: s,
    error: l,
    id: o,
    required: f
  }, /* @__PURE__ */ r.createElement(h, {
    spacing: 1
  }, i && /* @__PURE__ */ r.createElement(y, {
    action: a
  }, i), /* @__PURE__ */ r.createElement(R, {
    actions: n,
    label: i,
    nextLabel: m,
    onNext: u,
    onPrevious: p,
    previousLabel: c,
    secondaryLabel: g,
    selectedSlide: q,
    id: o,
    ...v
  }, d), /* @__PURE__ */ r.createElement(L, null), /* @__PURE__ */ r.createElement(F, null)));
};
t.defaultProps = {
  actions: void 0,
  error: void 0,
  hint: void 0,
  id: void 0,
  labelAction: void 0,
  required: !1,
  secondaryLabel: void 0
};
t.propTypes = {
  actions: e.node,
  children: e.node.isRequired,
  error: e.string,
  hint: e.oneOfType([e.string, e.node, e.arrayOf(e.node)]),
  id: e.string,
  label: e.string.isRequired,
  labelAction: e.element,
  nextLabel: e.string.isRequired,
  onNext: e.func.isRequired,
  onPrevious: e.func.isRequired,
  previousLabel: e.string.isRequired,
  required: e.bool,
  secondaryLabel: e.string,
  selectedSlide: e.number.isRequired
};
export {
  t as CarouselInput
};
