// @strapi/design-system/dist/Checkbox/Checkbox.js
import r from "react";
import c from "styled-components";
import e from "prop-types";
import { BaseCheckbox as s } from "../BaseCheckbox/BaseCheckbox.js";
import { Stack as f } from "../Stack/Stack.js";
import { Field as b } from "../Field/Field.js";
import "../Field/FieldLabel.js";
import "../Field/FieldInput.js";
import { FieldHint as u } from "../Field/FieldHint.js";
import { FieldError as x } from "../Field/FieldError.js";
import { useField as y } from "../Field/FieldContext.js";
import "../Field/FieldAction.js";
import { Box as h } from "../Box/Box.js";
import { useId as E } from "../helpers/useId.js";
import { Typography as g } from "../Typography/Typography.js";
const k = c(g)`
  display: flex;
  align-items: flex-start;
  * {
    cursor: ${({ disabled: o }) => o ? "not-allowed" : "pointer"};
  }
`, T = (o) => {
  const { id: t } = y();
  return /* @__PURE__ */ r.createElement(s, {
    id: t,
    ...o
  });
}, a = ({ children: o, disabled: t, id: p, hint: m, error: l, ...d }) => {
  const i = E("checkbox", p);
  let n;
  return l ? n = `${i}-error` : m && (n = `${i}-hint`), /* @__PURE__ */ r.createElement(b, {
    id: i,
    hint: m,
    error: l
  }, /* @__PURE__ */ r.createElement(f, {
    spacing: 1
  }, /* @__PURE__ */ r.createElement(k, {
    as: "label",
    textColor: "neutral800",
    disabled: t
  }, /* @__PURE__ */ r.createElement(T, {
    disabled: t,
    "aria-describedby": n,
    ...d
  }), /* @__PURE__ */ r.createElement(h, {
    paddingLeft: 2
  }, o)), /* @__PURE__ */ r.createElement(u, null), /* @__PURE__ */ r.createElement(x, null)));
};
a.defaultProps = {
  disabled: !1,
  id: void 0,
  error: void 0,
  hint: void 0
};
a.propTypes = {
  children: e.node.isRequired,
  disabled: e.bool,
  error: e.string,
  hint: e.oneOfType([e.string, e.node, e.arrayOf(e.node)]),
  id: e.oneOfType([e.string, e.number])
};
export {
  a as Checkbox
};
