// @strapi/design-system/dist/Button/utils.cjs
"use strict";Object.defineProperties(exports,{__esModule:{value:!0},[Symbol.toStringTag]:{value:"Module"}});const c=require("../Typography/Typography.cjs"),l=require("./constants.cjs"),s=o=>o===l.SUCCESS_LIGHT||o===l.DANGER_LIGHT?`${o.substring(0,o.lastIndexOf("-"))}`:o===l.TERTIARY?"neutral":o===l.DEFAULT||o===l.SECONDARY||l.VARIANTS.every(r=>r!==o)?"primary":`${o}`,u=({theme:o})=>`
    border: 1px solid ${o.colors.neutral200};
    background: ${o.colors.neutral150};
    ${c.Typography} {
      color: ${o.colors.neutral600};
    }
    svg {
      > g, path {
        fill: ${o.colors.neutral600};
      }
    }
  `,$=({theme:o,variant:r})=>[...l.LIGHT_VARIANTS,l.SECONDARY].includes(r)?`
      background-color: ${o.colors.neutral0};
    `:r===l.TERTIARY?`
      background-color: ${o.colors.neutral100};
    `:r===l.GHOST?`
      background-color: ${o.colors.neutral100};
    `:r===l.DEFAULT?`
      border: 1px solid ${o.colors.buttonPrimary500};
      background: ${o.colors.buttonPrimary500};
    `:`
    border: 1px solid ${o.colors[`${s(r)}500`]};
    background: ${o.colors[`${s(r)}500`]};
  `,n=({theme:o,variant:r})=>[...l.LIGHT_VARIANTS,l.SECONDARY].includes(r)?`
      background-color: ${o.colors.neutral0};
      border: 1px solid ${o.colors[`${s(r)}600`]};
      ${c.Typography} {
        color: ${o.colors[`${s(r)}600`]};
      }
      svg {
        > g, path {
          fill: ${o.colors[`${s(r)}600`]};
        }
      }
    `:r===l.TERTIARY?`
      background-color: ${o.colors.neutral150};
    `:`
    border: 1px solid ${o.colors[`${s(r)}600`]};
    background: ${o.colors[`${s(r)}600`]};
  `,e=({theme:o,variant:r})=>{switch(r){case l.DANGER_LIGHT:case l.SUCCESS_LIGHT:case l.SECONDARY:return`
          border: 1px solid ${o.colors[`${s(r)}200`]};
          background: ${o.colors[`${s(r)}100`]};
          ${c.Typography} {
            color: ${o.colors[`${s(r)}700`]};
          }
          svg {
            > g, path {
              fill: ${o.colors[`${s(r)}700`]};
            }
          }
        `;case l.TERTIARY:return`
          border: 1px solid ${o.colors.neutral200};
          background: ${o.colors.neutral0};
          ${c.Typography} {
            color: ${o.colors.neutral800};
          }
          svg {
            > g, path {
              fill: ${o.colors.neutral800};
            }
          }
        `;case l.GHOST:return`
        border: 1px solid transparent;
        background: transparent;

        ${c.Typography} {
          color: ${o.colors.neutral800};
        }

        svg {
          > g, path {
            fill: ${o.colors.neutral500};
          }
        }
      `;case l.SUCCESS:case l.DANGER:return`
          border: 1px solid ${o.colors[`${s(r)}600`]};
          background: ${o.colors[`${s(r)}600`]};
          ${c.Typography} {
            color: ${o.colors.neutral0};
          }
        `;default:return`
          svg {
            > g, path {
              fill: ${o.colors.buttonNeutral0};
            }
          }
        `}};exports.getActiveStyle=n;exports.getDisabledStyle=u;exports.getHoverStyle=$;exports.getVariantColorName=s;exports.getVariantStyle=e;
