// @strapi/design-system/dist/BaseRadio/BaseRadio.cjs
"use strict";Object.defineProperties(exports,{__esModule:{value:!0},[Symbol.toStringTag]:{value:"Module"}});const f=require("react/jsx-runtime"),a=require("react"),b=require("styled-components"),g=require("./context.cjs"),t=require("./utils.cjs"),r=e=>e&&typeof e=="object"&&"default"in e?e:{default:e},y=r(a),R=r(b),m=R.default.input`
  margin: 0;
  padding: 0;
  background-color: ${({theme:e})=>e.colors.neutral0};
  border: 1px solid ${({theme:e})=>e.colors.primary600};
  border-radius: 50%;
  height: ${({size:e})=>typeof e=="number"?e:t.getRadioSize(e)};
  width: ${({size:e})=>typeof e=="number"?e:t.getRadioSize(e)};
  -webkit-appearance: none;

  &:after {
    border-radius: 50%;
    content: '';
    position: relative;
    z-index: 1;
    display: block;
    height: ${({size:e})=>typeof e=="number"?e:t.getSelectedRadioSize(e)};
    width: ${({size:e})=>typeof e=="number"?e:t.getSelectedRadioSize(e)};
    left: ${t.getSelectedRadioPosition};
    top: ${t.getSelectedRadioPosition};
  }

  &:checked:after {
    background: ${({theme:e})=>e.colors.primary600};
  }

  &:disabled {
    border: 1px solid ${({theme:e})=>e.colors.neutral300};
    background: ${({theme:e})=>e.colors.neutral200};
  }
`,d=y.default.forwardRef(({value:e,disabled:n=!1,...i},c)=>{const{onChange:s,selected:l,name:u,size:p}=a.useContext(g.RadioContext),o=l===e;return f.jsx(m,{ref:c,type:"radio",name:u,value:e,tabIndex:o?0:-1,"aria-checked":o,checked:o,disabled:n,size:p,onChange:s,...i})});d.displayName="Radio";exports.BaseRadio=d;
