// @strapi/design-system/dist/Breadcrumbs/Breadcrumbs.js
import { jsxs as i, jsx as e } from "react/jsx-runtime";
import d from "styled-components";
import { ChevronRight as m } from "@strapi/icons";
import { Typography as o } from "../Typography/Typography.js";
import { Box as t } from "../Box/Box.js";
import { Flex as l } from "../Flex/Flex.js";
import { VisuallyHidden as p } from "../VisuallyHidden/VisuallyHidden.js";
const s = d(l)`
  svg {
    height: ${10 / 16}rem;
    width: ${10 / 16}rem;
    path {
      fill: ${({
  theme: r
}) => r.colors.neutral500};
    }
  }
  :last-of-type ${t} {
    display: none;
  }
  :last-of-type ${o} {
    color: ${({
  theme: r
}) => r.colors.neutral800};
    font-weight: ${({
  theme: r
}) => r.fontWeights.bold};
  }
`, h = ({
  children: r
}) => i(s, {
  inline: !0,
  as: "li",
  children: [e(o, {
    variant: "pi",
    textColor: "neutral600",
    children: r
  }), e(t, {
    "aria-hidden": !0,
    paddingLeft: 3,
    paddingRight: 3,
    children: e(m, {})
  })]
});
h.displayName = "Crumb";
const c = ({
  children: r,
  label: n,
  ...a
}) => i(l, {
  ...a,
  children: [e(p, {
    children: n
  }), e("ol", {
    "aria-hidden": !0,
    children: r
  })]
});
c.displayName = "Breadcrumbs";
export {
  c as Breadcrumbs,
  h as Crumb
};
