// @strapi/design-system/dist/Avatar/Avatar.cjs
"use strict";Object.defineProperties(exports,{__esModule:{value:!0},[Symbol.toStringTag]:{value:"Module"}});const r=require("react/jsx-runtime"),l=require("react"),u=require("styled-components"),c=require("../Typography/Typography.cjs"),d=require("../Flex/Flex.cjs"),t=require("./constants.cjs"),p=e=>e&&typeof e=="object"&&"default"in e?e:{default:e},i=p(u),h=i.default.img`
  border-radius: 50%;
  object-fit: cover;
  display: block;
  position: relative;
`,v=i.default.div`
  position: relative;
  width: ${t.avatarSize/16}rem;
  height: ${t.avatarSize/16}rem;
  z-index: ${({hovering:e})=>e?1:void 0};
`,x=i.default.img`
  border-radius: 50%;
  object-fit: cover;
  position: absolute;
  transform: translate(-${(t.previewSize-t.avatarSize)/2}px, -100%);
  margin-top: -${({theme:e})=>e.spaces[1]};
`,g=i.default.div`
  z-index: 1;
  border-radius: 30%;
  position: absolute;
  width: ${t.avatarSize/16}rem;
  height: ${t.avatarSize/16}rem;
  background: ${({theme:e})=>e.colors.neutral0};
  opacity: 0.4;
`,f=({src:e,alt:s,preview:a})=>{const[o,n]=l.useState(!1);return r.jsxs("span",{children:[a&&o?r.jsx(x,{"aria-hidden":!0,alt:"",width:`${t.previewSize}px`,height:`${t.previewSize}px`,src:a===!0?e:a}):null,r.jsxs(v,{hovering:Boolean(a&&o),onMouseEnter:()=>n(!0),onMouseLeave:()=>n(!1),children:[a&&o?r.jsx(g,{}):null,r.jsx(h,{src:e,alt:s,width:`${t.avatarSize}px`,height:`${t.avatarSize}px`})]})]})},y=i.default(d.Flex)`
  span {
    line-height: 0;
  }
`,b=({children:e,background:s="primary600",textColor:a="buttonNeutral0"})=>r.jsx(y,{background:s,borderRadius:"50%",width:`${t.avatarSize}px`,height:`${t.avatarSize}px`,justifyContent:"center",children:r.jsx(c.Typography,{fontSize:0,fontWeight:"bold",textColor:a,textTransform:"uppercase",children:e})});exports.Avatar=f;exports.Initials=b;
