// @strapi/design-system/dist/BaseButton/BaseButton.cjs
"use strict";Object.defineProperties(exports,{__esModule:{value:!0},[Symbol.toStringTag]:{value:"Module"}});const n=require("react/jsx-runtime"),l=require("react"),d=require("styled-components"),i=require("../themes/utils.cjs"),t=e=>e&&typeof e=="object"&&"default"in e?e:{default:e},c=t(l),p=t(d),s=p.default.button`
  display: flex;
  cursor: pointer;
  padding: ${({theme:e})=>e.spaces[2]};
  border-radius: ${({theme:e})=>e.borderRadius};
  background: ${({theme:e})=>e.colors.neutral0};
  border: 1px solid ${({theme:e})=>e.colors.neutral200};
  svg {
    height: ${({theme:e})=>e.spaces[3]};
    width: ${({theme:e})=>e.spaces[3]};
  }
  svg {
    > g,
    path {
      fill: ${({theme:e})=>e.colors.neutral0};
    }
  }
  &[aria-disabled='true'] {
    pointer-events: none;
  }

  ${i.buttonFocusStyle}
`,r=c.default.forwardRef(({disabled:e,children:o,...a},u)=>n.jsx(s,{ref:u,"aria-disabled":e,type:"button",disabled:e,...a,children:o}));r.displayName="BaseButton";exports.BaseButton=r;exports.BaseButtonWrapper=s;
